### PR TITLE
ROC-3226 add daily interest amount

### PR DESCRIPTION
**Pull request by robertparkinson**
### JIRA link
https://tools.hmcts.net/jira/browse/ROC-3226

this is a branch of my other story  as it requires those changes to work:
https://github.com/hmcts/cmc-citizen-frontend/pull/449

### Change description
Allow the user to give an amount of interest owed. How they came to that amount and the ability to add a specified amount of money daily.

### Work checklist

- [ ] Unit tests added where applicable
- [ ] Route tests added for new pages
- [ ] New pages included in a11y tests
- [ ] Task list and task completeness checks updated
- [ ] Check and send page updated
- [ ] Routes, page content and page flows of new features are toggled 
- [ ] UI changes look good on mobile
- [ ] Required Google Analytics events are being sent 

### Developer self-QA run statement

- [x] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

### CODE REVIEW

File: src/main/features/claim/form/models/interestHowMuch.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+  static readonly RATE_REQUIRED: string = 'You haven’t entered a daily amount'
+  static readonly RATE_NOT_VALID: string = 'Correct the amount you’ve entered'
+}
+
+export class InterestHowMuch implements CompletableTask {
+
+  @IsDefined({ message: ValidationErrors.TYPE_REQUIRED })
+  @IsIn(InterestRateOption.all(), { message: ValidationErrors.TYPE_REQUIRED })
+  type?: string
```
Comment (timja-kainos) [inline src/main/features/claim/form/models/interestHowMuch.ts]: this should have a type of `InterestRateOption`

File: src/main/features/claim/form/models/interestHowMuch.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  @ValidateIf(o => o.type === InterestRateOption.DIFFERENT)
+  @IsDefined({ message: ValidationErrors.RATE_REQUIRED })
+  @IsPositive({ message: ValidationErrors.RATE_NOT_VALID })
+  dailyAmount?: number
```
Comment (timja-kainos) [inline src/main/features/claim/form/models/interestHowMuch.ts]: does this need a Fractions validation on it?

File: src/main/features/claim/form/models/interestTotal.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+export class ValidationErrors {
+  static readonly REASON_REQUIRED: string = 'You must tell us how you calculated the amount'
+}
+
+export class InterestTotal implements CompletableTask {
+
+  @Min(0.01, { message: CommonValidationErrors.AMOUNT_NOT_VALID })
+  @Fractions(0, 2, { message: CommonValidationErrors.AMOUNT_INVALID_DECIMALS })
+  @IsDefined({ message: CommonValidationErrors.AMOUNT_REQUIRED })
```
Comment (timja-kainos) [inline src/main/features/claim/form/models/interestTotal.ts]: I think if you moved up the `IsDefined` up one you might not need to change the `Fractions` validator to accept `''`

File: src/main/features/claim/views/check-and-send.njk [hunk 4]
```diff
@@ -38,57 +38,80 @@
...
     }}
 
-    {{ row('Email:', draftClaim.defendant.email.address, paths.defendantEmailPage.uri, bold = true) }}
+    {{ row('Email', draftClaim.defendant.email.address, paths.defendantEmailPage.uri, bold = true) }}
     {{ tableEnd() }}
 
     {{ tableStart('claimAmount', 'Claim amount') }}
-    {{ row('Claim amount:', claimAmountTotal.claimAmount | numeral, paths.amountPage.uri, bold = true) }}
+
+    {{ row('Claim amount breakdown', '', paths.amountPage.uri, bottomBorder = false, bold = true) }}
```
Comment (timja-kainos) [inline src/main/features/claim/views/check-and-send.njk]: these check and send content changes should be in a separate PR, they could get merged very quickly

File: src/main/features/claim/views/interest-continue-claiming.njk [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    <div class="column-two-thirds">
+
+      {% set name = 'option' %}
+
+      <form novalidate method="post">
+        {{ csrfProtection(csrf) }}
+
+        {{
+        radioGroup(
+          label = 'Continue to claim interest after you submit your claim?',
```
Comment (timja-kainos) [inline src/main/features/claim/views/interest-continue-claiming.njk]: heading and label should be in a variable really so they stay in sync

you could just use the `heading` variable here

File: src/main/features/claim/views/interest-how-much.njk [hunk 6]
```diff
@@ -0,0 +1,47 @@
+{% extends "layout.njk" %}
+{% from "form.njk" import csrfProtection, poundInput, radioOption, saveAndContinueButton, errorSummary %}
+
+{% set title = t('Money claim') %}
+{% set heading = t('How much do you want to continue claiming?') %}
```
Comment (timja-kainos) [inline src/main/features/claim/views/interest-how-much.njk]: same here

File: src/test/features/claim/form/models/interestContinueClaiming.ts [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+describe('InterestContinueClaiming', () => {
+
+  describe('form object deserialization', () => {
+
+    it('should return undefined when value is undefined', () => {
+      expect(InterestContinueClaiming.fromObject(undefined)).to.be.equal(undefined)
+    })
+
+    it('should return null when value is null', () => {
```
Comment (timja-kainos) [inline src/test/features/claim/form/models/interestContinueClaiming.ts]: we don't use null so don't need to test against null, remove please
REQ

File: src/test/features/claim/form/models/interestHowMuch.ts [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+
+describe('InterestHowMuch', () => {
+
+  describe('form object deserialization', () => {
+
+    it('should return undefined when value is undefined', () => {
+      expect(InterestHowMuch.fromObject(undefined)).to.be.equal(undefined)
+    })
+
+    it('should return null when value is null', () => {
```
Comment (timja-kainos) [inline src/test/features/claim/form/models/interestHowMuch.ts]: same here REQ
